Dashboard/login.py

```diff
@@ -6,51 +6,69 @@
 from time import sleep
 
 import streamlit as st
+from bcrypt import gensalt, hashpw, checkpw
 from dotenv import load_dotenv
 from psycopg2.extensions import connection
 
 from helpers import get_connection
 from navigation import make_sidebar
 from custom_styling import apply_custom_styling
 
-EMAIL_PATTERN = r"""(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])"""
+EMAIL_PATTERN = r"""(?:[a-z0-9!#$%&'*+/=?^_`{|}~-]+(?:\.[a-z0-9!#$%&'*+/=?^_`{|}~-]+)*|\"(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21\x23-\x5b\x5d-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])*\")@(?:(?:[a-z0-9](?:[a-z0-9-]*[a-z0-9])?\.)+[a-z0-9](?:[a-z0-9-]*[a-z0-9])?|\[(?:(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9]))\.){3}(?:(2(5[0-5]|[0-4][0-9])|1[0-9][0-9]|[1-9]?[0-9])|[a-z0-9-]*[a-z0-9]:(?:[\x01-\x08\x0b\x0c\x0e-\x1f\x21-\x5a\x53-\x7f]|\\[\x01-\x09\x0b\x0c\x0e-\x7f])+)\])"""  # pylint: disable=line-too-long
 
-def authenticate(conn: connection, email: str) -> tuple | None:
+
+def get_email(conn: connection, email: str) -> tuple[str, str] | None:
     """Find email in database"""
     logging.info("Searching for %s in database", email)
     with conn.cursor() as cur:
         cur.execute("""
-            SELECT email FROM users
+            SELECT email, CONVERT_FROM(password, 'UTF8') FROM users
             WHERE email = %s""", (email,))
         data = cur.fetchone()
     return data
 
 
-def add_email(conn: connection, email: str) -> tuple | None:
+def add_email(conn: connection, email: str, password: bytes) -> tuple | None:
     """Add email to database"""
     logging.info("Adding %s to database", email)
     with conn.cursor() as cur:
         cur.execute("""
-            INSERT INTO users(email)
-            VALUES (%s)
-            RETURNING email""", (email,))
+            INSERT INTO users(email, password)
+            VALUES (%s, %s)
+            RETURNING *""", (email, password))
         data = cur.fetchone()
         conn.commit()
     return data
 
 
+def login(email: str) -> None:
+    """Login and move to next page"""
+    st.session_state.logged_in = True
+    st.session_state.email = email
+    logging.info("Welcome! You are logged in with email: %s", email)
+    st.success(f"Welcome! You are logged in with email: {email}")
+    sleep(0.5)
+    st.switch_page("pages/price.py")
+
+
 def login_page(config: _Environ, email_pattern: str) -> None:
     """Login page"""
-    st.header("Sign in")
```

Comment:
Better to be consistent with websites which normally say "Log in" and it's more confusing because they sound similar and the code already refers to it as login